api/base.go

```diff
@@ -255,6 +256,19 @@ func (h *BaseHandler) modBody(
 	return
 }
 
+//
+// CurrentUsername gets username from Keycloak auth token
+func (h *BaseHandler) CurrentUsername(ctx *gin.Context) (username string) {
+	token := ctx.GetHeader("Authorization")
+	// Consider later add username to ctx and cache for single request?
+	username, err := h.AuthProvider.Username(token)
```

Comment:
Since the middleware has already authenticated the token - how can this fail?

Reply:
This can fail when the auth token doesn't contain a field with the username. It looks to me to be responsibility of auth pkg to report it as an error, but a responsibility of the api pkg to decide what to do with missing username (setting it to "").